pyproject.toml

```diff
@@ -71,7 +72,7 @@ docs = [
 ]
 lint = [
     "codecov-cli",
-    "mypy",
+    "mypy == 1.11.2",
```

Comment:
An attempt to get CI working consistently with my local, which didn't work. But at the same time, mypy is one thing that really **should** be pinned or else individual pr's performing re-locks are going to randomly update it and get breaking linting changes.